### PR TITLE
replacing getlogin C call with a wrap

### PR DESCRIPTION
**Pull request by Matthieu-Sauer**
this is a fix of issue #74 . the crash was caused by getlogin not returning anything on my system.

### CODE REVIEW

**Review by vaxerski — COMMENTED** [review]
lgtm but needs a clang-format

**Comment by Matthieu-Sauer** [comment]
should be done

File: src/renderer/widgets/IWidget.cpp [hunk 1]
```diff
@@ -58,13 +59,13 @@ static std::string getTime() {
...
 
-    char* username = getlogin();
+    auto username = getUserName();
 
     if (!username)
         Debug::log(ERR, "Error in formatString, username null. Errno: ", errno);
 
     IWidget::SFormatResult result;
-    replaceAll(in, "$USER", std::string{username ? username : ""});
+    replaceAll(in, "$USER", username ? *username : std::string{""});
```
Comment (vaxerski) [inline src/renderer/widgets/IWidget.cpp:68]: `getUserName().value_or("")`?

Comment (Matthieu-Sauer) [reply src/renderer/widgets/IWidget.cpp]: yeah fixed (i'm so dumb)

**Comment by vaxerski** [comment]
conflict to be fixed, btw

**Pull request closed without merging**